### PR TITLE
Improves interactions with display cases.

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Adds much better feedback for interacting with and examining display cases.
Clicking on an open display case will now try to put the item into your hands if possible, or drop the item onto the ground if it can't.

## Why It's Good For The Game
I've personally been noob-trapped by these before; it's not always clear how you're supposed to interact with this item, which is especially annoying for something that frequently holds onto a high-value item.

## Testing
Busted up a few cases, and checked them out.

## Changelog
:cl:
tweak: Display cases are now a bit more informative on examine and interaction.
tweak: Items removed from display cases will now be placed into your hand.
/:cl:

### CODE REVIEW

**Comment by 1080pCat** [comment]
> I've personally been noob-trapped by these before; it's not always clear how you're supposed to interact with this item,

i ded pls buff :pog:

File: code/game/objects/structures/displaycase.dm [hunk 1]
```diff
@@ -46,18 +49,32 @@
...
-		. += "<span class='warning'>The ID lock has been shorted out.</span>"
 	if(showpiece)
-		. += "<span class='notice'>There's [showpiece] inside.</span>"
+		. += "<span class='notice'>There's \a [showpiece] displayed inside.</span>"
+	else
+		. += "<span class='notice'>It's empty.</span>"
 	if(trophy_message)
 		. += "The plaque reads:\n [trophy_message]"
+	if(!openable)
+		. += "<span class='notice'>It seems to be sealed shut, there's no way you're getting that open.</span>"
```
Comment (1080pCat) [inline code/game/objects/structures/displaycase.dm:59]: When/where does this ever come up?

File: code/game/objects/structures/displaycase.dm [hunk 2]
```diff
@@ -46,18 +49,32 @@
...
+	else
+		if(!open)
+			. += "<span class='notice'>The ID lock is active, you need to swipe an ID to open it.</span>"
+		else if((broken || open) && showpiece)
+			. += "<span class='notice'>[showpiece] is held up by a holo-field. You can deactivate the field with an open hand.</span>"
```
Comment (1080pCat) [inline code/game/objects/structures/displaycase.dm:64]:
```suggestion
			. += "<span class='notice'>[showpiece] is held in a loose low gravity suspension feild. You can deactivate the field with an open hand.</span>"
```

**Review by 1080pCat — APPROVED** [review]
looks good to mee

File: code/game/objects/structures/displaycase.dm (hunk 2 above)
Comment (DGamerL) [reply code/game/objects/structures/displaycase.dm:64]:
```suggestion
			. += "<span class='notice'>[showpiece] is held in a loose low gravity suspension field. You can deactivate the field with an open hand.</span>"
```

Comment (1080pCat) [reply code/game/objects/structures/displaycase.dm:64]: woops

File: code/game/objects/structures/displaycase.dm [hunk 3]
```diff
@@ -207,7 +236,7 @@
 		to_chat(user, "<span class='notice'>You start installing the electronics into [src]...</span>")
 		playsound(loc, I.usesound, 50, 1)
```
Comment (Burzah) [inline code/game/objects/structures/displaycase.dm:237]:
```suggestion
		playsound(loc, I.usesound, 50, TRUE)
```
While we are here.

File: code/game/objects/structures/displaycase.dm [hunk 4]
```diff
@@ -46,18 +55,30 @@
...
+		else if((broken || open) && showpiece)
+			. += "<span class='notice'>[showpiece] is held in a loose low gravity suspension field. You can take [showpiece] out[broken ? "." : ", or lock [src] with an ID"].</span>"
+
+	if(alert)
+		. += "<span class='notice'>It is hooked up with an anti-theft system.</span>"
+	if(emagged)
+		. += "<span class='warning'>The ID lock has been shorted out.</span>"
 
-/obj/structure/displaycase/proc/dump()
+
```
Comment (DGamerL) [inline code/game/objects/structures/displaycase.dm:77]:
```suggestion
```

**Review by DGamerL — APPROVED** [review]

File: code/game/objects/structures/displaycase.dm [hunk 5]
```diff
@@ -109,9 +130,15 @@
...
+	if(I.GetID())
+		if(!openable)
+			to_chat(user, "<span class='warning'>There is no ID scanner, looks like this one is sealed shut.</span>")
+			return
+		if(broken)
+			to_chat(user, "<span class='warning'>[src] is broken, the ID lock won't do anything.</span>")
+			return
 		if(allowed(user) || emagged)
-			to_chat(user, "<span class='notice'>You [open ? "close":"open"] [src].</span>")
+			to_chat(user, "<span class='notice'>You use [I] to [open ? "close":"open"] [src].</span>")
```
Comment (Henri215) [inline code/game/objects/structures/displaycase.dm:141]:
```suggestion
			to_chat(user, "<span class='notice'>You use [I] to [open ? "close" : "open"] [src].</span>")
```

**Review by Henri215 — APPROVED** [review]
Looks good to Oyu

**Review by Burzah — APPROVED** [review]

**Pull request merged**